### PR TITLE
added project structure

### PR DESCRIPTION
**Pull request by Sheelendar196**

### CODE REVIEW

File: src/graphql-project/internal/core/domain/employee.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	LastUpdatedBy *string    `validate:"omitempty,ascii,max=100"`
+}
+
+func GetEmployee(name, empID, mobile, email, department,
+	add, managerID string, isActive bool) *Employee {
+
+	return &Employee{
+		Name:       name,
+		EmpID:      empID,
+		Mobile:     &mobile,
```
Comment (sunitha-premakumaran) [inline src/graphql-project/internal/core/domain/employee.go]: These can be passed a pointer to the function?

File: src/graphql-project/internal/infrastructure/repository/employee.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	}
+
+	return nil, errors.New("employee not found in db")
+}
+
+func (e *Employee) GetEmployeeList(ctx context.Context) ([]*domain.Employee, error) {
+	return e.list, nil
+}
+
+func (e *Employee) UpdateEmployeeDetails(ctx context.Context, input domain.Employee) (*domain.Employee, error) {
```
Comment (sunitha-premakumaran) [inline src/graphql-project/internal/infrastructure/repository/employee.go:44]: These logic checks should happen in worker. Keep only the calls to the db in this layer

File: src/graphql-project/internal/infrastructure/repository/employee.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	}
+	if index == -1 {
+		return nil, errors.New("no employee found with this empID")
+	}
+	emp := es.list[index]
+	es.list = append(es.list[:index], es.list[index+1:]...)
+	return emp, nil
+}
+
+func updateEmployee(emp *domain.Employee, input domain.Employee) {
```
Comment (sunitha-premakumaran) [inline src/graphql-project/internal/infrastructure/repository/employee.go]: Keep the table model and domain datatypes same

File: src/graphql-project/internal/infrastructure/repository/employee.go (hunk 2 above)
Comment (Sheelendar196) [reply src/graphql-project/internal/infrastructure/repository/employee.go:44]: UpdateEmployeeDetails update employee into table on once actual table will be there this logic removed. it is just search into table(list)

File: src/graphql-project/internal/infrastructure/repository/employee.go (hunk 3 above)
Comment (Sheelendar196) [reply src/graphql-project/internal/infrastructure/repository/employee.go]: ok done

File: src/graphql-project/internal/core/domain/employee.go (hunk 1 above)
Comment (Sheelendar196) [reply src/graphql-project/internal/core/domain/employee.go]: removed not required

**Pull request merged**